### ISSUE TITLE
Streaming: failure to drain output stream unless <out-stream>.end() is called

### ISSUE THREAD

**Issue by jldec** [projectriff/node-function-invoker#164]
Testing with current v0.6.0-snapshot release bits

The function below writes an output message (the current running average) when an input is negative, and re-computes a running average as long as the the inputs are positive.

With the function as written below, and the input stream `on('end',...)` handler disabled, the output stream is not drained, and nothing appears on the output stream after publishing a positive and a negative value to trigger the output. This is unexpected.

The function does work when the `on('end')` handler is uncommented, to call `.end()` on the output stream.

```js
module.exports = (inputStreams, outputStreams) => {
  const ins = inputStreams.in;
  const outs  = outputStreams.out;
  console.log('new invocation');

  let avg = undefined;
  let sum = 0;
  let cnt = 0;

  ins.on('data', (x) => {

    if (x < 0) {
      if (avg) {
        let rval = outs.write(avg);
        console.log('outs.write: ', avg, rval ? '(OK)' : '(not OK)');
      }
      avg = undefined;
      sum = 0;
      cnt = 0;
      return;
    }

    if (x >= 0) {
      sum += x;
      cnt ++;
      avg = sum/cnt;
      console.log('avg updated: ', avg);
    }
  });

  ins.on("end", () => {
    // *********************************************
    // outs.end();
    // console.log('end on ins, call outs.end()');
    // *********************************************
    console.log('end on ins, NO outs.end()');
  });
  ins.on("error", (err) => {
    outs.end();
    console.log('error on ins, outs.end()', err);
  });
  outs.on("error", (err) => {
    ins.end();
    console.log('error on outs, ins.end()', err);
  });
};

module.exports.$interactionModel = 'node-streams';
```

**Comment by fbiville** [comment]
TL;DR: I cannot reproduce the issue

I tried to reproduce the issue with a simpler version of the program, i.e. rid of end/error handlers:

```js
module.exports = (inputStreams, outputStreams) => {
    const { ins } = inputStreams;
    const { outs } = outputStreams;
    let avg = undefined;
    let sum = 0;
    let cnt = 0;
    ins.on("data", x => {
        if (x < 0) {
            if (avg) {
                outs.write(avg);
                console.log("output: ", avg);
            }
            avg = undefined;
            sum = 0;
            cnt = 0;
            return;
        }
        if (x >= 0) {
            sum += x;
            cnt++;
            avg = sum / cnt;
            console.log("average updated: ", avg);
        }
    });
};
module.exports.$interactionModel = "node-streams";
```

I ran this with both the invoker and streaming processor running on my machine (as well as a simple main function wrapping `stream-client-go` to send data to remote streams).
I got the expected `"average updated: "` logs whenever a positive number was sent, as well as the expected `"output"` when I sent a negative number.

**Comment by fbiville** [comment]
Closed due to lack of reproducibility.
